model2vec/distill/tokenizer.py

```diff
@@ -68,7 +68,17 @@ def remove_tokens(tokenizer: Tokenizer, tokens_to_remove: list[str]) -> Tokenize
         tokenizer_data["model"]["vocab"] = reindexed
 
     elif model_type == "Unigram":
-        logger.warning("Removing tokens from a unigram tokenizer is not supported.")
+        # Vocab is a list of tuples.
+        vocab: list[dict[str, Any]] = tokenizer_data["model"]["vocab"]
+        n_tokens = len(vocab)
+        # Remove the tokens.
+        for token in tokens_to_remove:
```

Comment:
Consider turning this into a dictionary operation, you now do N passes over the vocabulary in the worst case. E.g., you could do it in single pass like so:

```python
token_remove_set = set(tokens_to_remove)
new_vocab = [(token, score) for token, score in vocab if token not in tokens_to_remove_set]
```